lib/shopify_transporter/pipeline/magento/product/top_level_attributes.rb

```diff
@@ -8,10 +8,26 @@ module Magento
       module Product
         class TopLevelAttributes < Pipeline::Stage
           def convert(hash, record)
+            warn_if_too_many_options(hash)
             accumulator = TopLevelAttributesAccumulator.new(record)
             accumulator.accumulate(hash)
           end
 
+          private
+
+          MAX_OPTION_COUNT = 3
```

Comment:
Minor: I suggest moving `MAX_OPTION_COUNT` to the top above convert because it ends up being public. Or you can declare it private by doing this:
```ruby
MAX_OPTION_COUNT = 3
private_constant :MAX_OPTION_COUNT
```